### PR TITLE
DFTHG# 77 Setup Skeleton for Driver Portal

### PR DESCRIPTION
**Pull request by shruthi-rapala**
-Added angular project for the drivers portal
- Added tests project to driver portal

### CODE REVIEW

File: driver-portal/src/Controllers/test.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.AspNetCore.Mvc;
+
+namespace MigrationMetrics.Controllers
```
Comment (GeorgeWalker) [inline driver-portal/src/Controllers/test.cs:3]: This namespace needs to be changed

Comment (shruthi-rapala) [reply driver-portal/src/Controllers/test.cs]: Adjusted the namespace and pushed the changes.

File: driver-portal/openshift/templates/driver-portal.template.yml [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    apiVersion: v1
+    type: Opaque
+    metadata:
+      name: ${APP_NAME}-driver-portal-api-secrets
+      namespace: ${NAMESPACE_PREFIX}-${ENV}
+      labels:
+        app: ${APP_NAME}-driver-portal
+        app.kubernetes.io/part-of: ${APP_NAME}
+    stringData:      
+      DATABASE_LOCATION: '/db/DATA.db'
```
Comment (GeorgeWalker) [inline driver-portal/openshift/templates/driver-portal.template.yml:67]: this is not relevant

File: driver-portal/openshift/templates/driver-portal.template.yml [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+    apiVersion: route.openshift.io/v1
+    metadata:
+      labels:
+        app: ${APP_NAME}-driver-portal
+        app.kubernetes.io/part-of: ${APP_NAME}
+      name: ${APP_NAME}-driver-portal-ui
+      namespace: ${NAMESPACE_PREFIX}-${ENV}
+    spec:
+      host: ${HOST}
+      path: /metrics
```
Comment (GeorgeWalker) [inline driver-portal/openshift/templates/driver-portal.template.yml:115]: set the path to /driver-portal

File: driver-portal/openshift/templates/driver-portal.template.yml [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+      labels:
+        app: ${APP_NAME}-migration-metrics
+        app.kubernetes.io/part-of: ${APP_NAME}
+    spec:
+      accessModes:
+        - "ReadWriteMany"
+      resources:
+        requests:
+          storage: "100Mi"
+  - kind: "PersistentVolumeClaim"
```
Comment (GeorgeWalker) [inline driver-portal/openshift/templates/driver-portal.template.yml:313]: This PVC is not needed

File: driver-portal/openshift/templates/driver-portal.template.yml [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+          dnsPolicy: ClusterFirst
+          restartPolicy: Always
+          schedulerName: default-scheduler
+          securityContext: {}
+          terminationGracePeriodSeconds: 30
+          volumes:
+            - name: dp-keys
+              persistentVolumeClaim:
+                claimName: ${APP_NAME}-driver-portal-api-key-ring
+            - name: db
```
Comment (GeorgeWalker) [inline driver-portal/openshift/templates/driver-portal.template.yml:209]: db is not required

File: driver-portal/src/ClientApp/Caddyfile [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+		# Set CSP to support Angular
+		Content-Security-Policy "default-src 'self' 'unsafe-eval' https://assets.phsa.ca https://fonts.gstatic.com https://*.gov.bc.ca https://*.bcgov https://localhost:5020; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com; script-src 'self' 'unsafe-inline' 'unsafe-eval'; frame-src https://phsaehealth-eformsweb.azurewebsites.net; frame-ancestors 'self' https://phsaehealth-eformsweb.azurewebsites.net;"
+		# Set referrer policy
+		Referrer-Policy "same-origin"
+	}
+
+	encode zstd gzip
+
+	route {
+		uri strip_prefix /rsbc-dfp-driver-portal
```
Comment (GeorgeWalker) [inline driver-portal/src/ClientApp/Caddyfile:27]: this parameter has to match the actual path in the route

File: driver-portal/src/ClientApp/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,17 @@
+FROM trion/ng-cli-karma AS ng-builder
+WORKDIR /src
+COPY ./package*.json ./
+RUN npm install
+COPY ./ .
+#RUN npm run lint
+#RUN npm run test -- --no-watch --no-progress
+
+RUN npm run build -- --base-href /driver-portal/ --deploy-url /driver-portal/ --configuration production
```
Comment (GeorgeWalker) [inline driver-portal/src/ClientApp/Dockerfile:9]: note that this path is correct - we do want it to hang off of /driver-portal

File: driver-portal/src/ClientApp/README.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Driver-Portal
+
+This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 12.0.2.
```
Comment (GeorgeWalker) [inline driver-portal/src/ClientApp/README.md:3]: bump the version to 15x

File: driver-portal/src/Program.cs [hunk 9]
```diff
@@ -0,0 +1,74 @@
+
+using System.Text.Json.Serialization;
+using Microsoft.OpenApi.Models;
+
+var builder = WebApplication.CreateBuilder(args);
+
```
Comment (GeorgeWalker) [inline driver-portal/src/Program.cs:6]: should be 

builder.WebHost
    .UseUrls()
    .UseKestrel(options =>
    {
    options.Listen(IPAddress.Any, 8080);
    });

File: driver-portal/openshift/templates/driver-portal.template.yml (hunk 2 above)
Comment (shruthi-rapala) [reply driver-portal/openshift/templates/driver-portal.template.yml]: Deleted all the references to DB in the openshift template

File: driver-portal/openshift/templates/driver-portal.template.yml (hunk 4 above)
Comment (shruthi-rapala) [reply driver-portal/openshift/templates/driver-portal.template.yml]: Removed the PVC claims for the DB

**Pull request merged**